ts/src/pro/cryptocom.ts

```diff
@@ -447,6 +451,97 @@ export default class cryptocom extends cryptocomRest {
             this.balance = this.safeBalance (this.balance);
         }
         client.resolve (this.balance, messageHash);
+        const messageHashRequest = this.safeString (message, 'id');
+        client.resolve (this.balance, messageHashRequest);
+    }
+
+    async createOrderWs (symbol: string, type: OrderType, side: OrderSide, amount: number, price: number = undefined, params = {}) {
+        /**
+         * @method
+         * @name cryptocom#createOrderWs
+         * @see https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#private-create-order
+         * @description create a trade order
+         * @param {string} symbol unified symbol of the market to create an order in
+         * @param {string} type 'market' or 'limit'
+         * @param {string} side 'buy' or 'sell'
+         * @param {float} amount how much of currency you want to trade in units of base currency
+         * @param {float} price the price at which the order is to be fullfilled, in units of the quote currency, ignored in market orders
+         * @param {object} [params] extra parameters specific to the cryptocom api endpoint
+         * @returns {object} an [order structure]{@link https://docs.ccxt.com/en/latest/manual.html#order-structure}
+         */
+        await this.loadMarkets ();
+        params = this.createOrderRequest (symbol, type, side, amount, price, params);
+        const request = {
+            'method': 'private/create-order',
+            'params': params,
+        };
+        const messageHash = this.nonce ();
+        return await this.watchPrivateRequest (messageHash, request);
+    }
+
+    handleOrder (client: Client, message) {
+        //
+        //    {
+        //        "id": 1,
+        //        "method": "private/create-order",
+        //        "code": 0,
+        //        "result": {
+        //            "client_oid": "c5f682ed-7108-4f1c-b755-972fcdca0f02",
+        //            "order_id": "18342311"
+        //        }
+        //    }
+        //
+        const messageHash = this.safeString (message, 'id');
+        const rawOrder = this.safeValue (message, 'result', {});
+        const order = this.parseOrder (rawOrder);
+        client.resolve (order, messageHash);
+    }
+
+    async cancelOrderWs (id: string, symbol: string = undefined, params = {}) {
+        /**
+         * @method
+         * @name cryptocom#cancelOrder
+         * @description cancels an open order
+         * @see https://exchange-docs.crypto.com/exchange/v1/rest-ws/index.html#private-cancel-order
+         * @param {string} id the order id of the order to cancel
+         * @param {string} [symbol] unified symbol of the market the order was made in
+         * @param {object} [params] extra parameters specific to the cryptocom api endpoint
+         * @returns {object} An [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        await this.loadMarkets ();
```

Comment:
any clue why I'm having this error?
```
 p cryptocom cancelOrderWs  "6142909895323193963"
Python v3.10.9
CCXT v4.0.10
cryptocom.cancelOrderWs(6142909895323193963)
Traceback (most recent call last):
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 239, in <module>
    asyncio.run(main())
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/opt/homebrew/Cellar/python@3.10/3.10.9/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/cjg/Git/ccxt10/ccxt/examples/py/cli.py", line 216, in main
    result = await method(*args)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 470, in cancel_order_ws
    return await self.watch_private_request(messageHash, request)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 513, in watch_private_request
    return await self.watch(url, str(nonce), message, True)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/pro/cryptocom.py", line 542, in handle_error_message
    self.throw_exactly_matched_exception(self.exceptions['exact'], errorCode, feedback)
  File "/Users/cjg/Git/ccxt10/ccxt/python/ccxt/async_support/base/exchange.py", line 2181, in throw_exactly_matched_exception
    raise exact[string](message)
ccxt.base.errors.BadRequest: cryptocom {"id":1688726256929,"method":"private/cancel-order","code":10004,"message":"BAD_REQUEST","original":"{\"id\":1688726256929,\"method\":\"private/cancel-order\",\"params\":{\"order_id\":6142909895323193963},\"nonce\":1688726256929}"}
```

Reply:
It was using the old exchange url instead of the unified url, I merged main with Dan's latest PR and should now work.